include/git2/worktree.h

```diff
@@ -148,6 +148,25 @@ GIT_EXTERN(int) git_worktree_unlock(git_worktree *wt);
  */
 GIT_EXTERN(int) git_worktree_is_locked(git_buf *reason, const git_worktree *wt);
 
+
+/**
+ * Retrieve the name of the worktree
+ *
+ * @param wt Worktree to get the name for
+ * @return The worktrees name or NULL for an error. The pointer returned
+ *  is valid for the lifetime of the git_worktree
```

Comment:
Nitpick: no leading " ".

About the message itself, I'd be tempted to not say anything about NULL being an error. AFAICT, having no idea what's the underlying implementation, a worktree with no name would be a pretty crazy thing to happen, so it should have been caught way earlier.